src/common/cuepoint-manager.js

```diff
@@ -0,0 +1,72 @@
+//@flow
+import {Cue, FakeEvent} from '@playkit-js/playkit-js';
+
+interface CuePoint {
+  id: string;
+  startTime: number;
+  endTime: number;
+}
+
+export class CuePointManager {
+  Event = {
+    TIMED_METADATA_ADDED: 'timedmetadataadded'
+  };
+  Type = {
+    CuePointsTextTrack: 'CuePoints',
+    CuePointKey: 'CuePoint'
+  };
+  _player: KalturaPlayer;
+  _textTrack: TextTrack | null = null;
+
+  constructor(player: KalturaPlayer) {
+    this._player = player;
```

Comment:
on reset need to clear cuepoints, they are media session dependent, so cuepoints of one media are not relevant for next media.
Listen to reset on kaltura player and call `clearAllCuePoints()`

On destroy need to do the same.